### ISSUE TITLE
Upgrading to sbt 0.6.1 breaks the protoc bridge

### ISSUE THREAD

**Issue by jvican** [scalapb/ScalaPB#314]
So you have this in your build:
```scala
addSbtPlugin("com.thesamet" % "sbt-protoc" % "0.99.12-rc4")
libraryDependencies += "com.trueaccord.scalapb" %% "compilerplugin" % "0.6.0"

```
  
And then you decide to bump up the minor version number of the scalapb dependency, changing `0.6.0` to `0.6.1`. This is what you get when you run `compile`.
```
protoc-jar: protoc version: 330, detected platform: linux/amd64
protoc-jar: executing: [/tmp/protocjar3638771974775296720/bin/protoc.exe, --plugin=protoc-gen-scala=/tmp/protocbridge2672480270281404332, --scala_out=grpc:/data/rw/code/scala/zinc/internal/zinc-persist/target/scala-2.12/src_managed/main, -I/data/rw/code/scala/zinc/internal/zinc-persist/src/main/protobuf, -I/data/rw/code/scala/zinc/internal/zinc-persist/target/protobuf_external, /data/rw/code/scala/zinc/internal/zinc-persist/src/main/protobuf/schema.proto]
java.lang.NoSuchFieldError: enumOptions
	at com.trueaccord.scalapb.compiler.DescriptorPimps$EnumDescriptorPimp.scalaOptions(DescriptorPimps.scala:378)
	at com.trueaccord.scalapb.compiler.DescriptorPimps$EnumDescriptorPimp.baseTraitExtends(DescriptorPimps.scala:409)
	at com.trueaccord.scalapb.compiler.ProtobufGenerator.printEnum(ProtobufGenerator.scala:23)
	at com.trueaccord.scalapb.compiler.ProtobufGenerator.$anonfun$generateMultipleScalaFilesForFileDescriptor$2(ProtobufGenerator.scala:1293)
	at com.trueaccord.scalapb.compiler.FunctionalPrinter.$anonfun$call$1(FunctionalPrinter.scala:60)
	at scala.collection.IndexedSeqOptimized.foldLeft(IndexedSeqOptimized.scala:56)
	at scala.collection.IndexedSeqOptimized.foldLeft$(IndexedSeqOptimized.scala:64)
	at scala.collection.mutable.WrappedArray.foldLeft(WrappedArray.scala:38)
	at com.trueaccord.scalapb.compiler.FunctionalPrinter.call(FunctionalPrinter.scala:60)
	at com.trueaccord.scalapb.compiler.ProtobufGenerator.$anonfun$generateMultipleScalaFilesForFileDescriptor$1(ProtobufGenerator.scala:1293)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at com.trueaccord.scalapb.compiler.ProtobufGenerator.generateMultipleScalaFilesForFileDescriptor(ProtobufGenerator.scala:1287)
	at com.trueaccord.scalapb.compiler.ProtobufGenerator$.$anonfun$handleCodeGeneratorRequest$3(ProtobufGenerator.scala:1354)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at com.trueaccord.scalapb.compiler.ProtobufGenerator$.handleCodeGeneratorRequest(ProtobufGenerator.scala:1349)
	at scalapb.ScalaPbCodeGenerator$.run(ScalaPbCodeGenerator.scala:16)
	at protocbridge.frontend.PluginFrontend$.$anonfun$runWithBytes$1(PluginFrontend.scala:56)
	at scala.util.Try$.apply(Try.scala:209)
	at protocbridge.frontend.PluginFrontend$.runWithBytes(PluginFrontend.scala:54)
	at protocbridge.frontend.PluginFrontend$.runWithInputStream(PluginFrontend.scala:67)
	at protocbridge.frontend.PosixPluginFrontend$.$anonfun$prepare$1(PosixPluginFrontend.scala:25)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:653)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

Before, in `0.6.0`, it was working.

**Comment by thesamet** [comment]
I can't reproduce this. From the error, it seems like it's trying to access a field that was added between 0.6.0 and 0.6.1. Is it possible that somehow you are getting a class file from compilerplugin 0.6.0 into your build?

**Comment by jvican** [comment]
sbt 1.0.0-RC2 depends on Zinc which depends on Protobuf, but we use the shaded compiler plugin. So, in theory, it *should not* be getting a class file from compilerplugin "0.6.0", right? This happens to me in the Zinc repository.

**Comment by thesamet** [comment]
No, what's shaded is Google's protobuf-java, not ScalaPB itself.

**Comment by xuwei-k** [comment]
https://github.com/xuwei-k/scalapb-314

**Comment by jvican** [comment]
Oops... there has been a misunderstanding then. Can you update the docs to reflect this info? The only reference to shading I saw is here: https://scalapb.github.io/scala.js.html and it doesn't say that only the Java protobuf compiler is shaded. I assumed everything was.
  
Is there any reason why are you not loading the whole scalapb dependency in a new classloader? It's the safest thing to do, and you will also remove the issue with scalajs compat. Otherwise, this makes any version bigger than 0.6.0 unusable with sbt 1.0.0.
  
A reference on how to load something in a new classloader: https://github.com/scalacenter/sbtfix/blob/master/sbt-rewrites-plugin/src/main/scala/sbt/rewrite/plugin/SbtMigrationPlugin.scala.

**Comment by xuwei-k** [comment]
workaround https://github.com/xuwei-k/scalapb-314/commit/2500965747186ec257e8b7c53f4fcff445636e95

**Comment by thesamet** [comment]
The problem should be fixed in ScalaPB >= 0.6.4.
